pydantic/decorators.py

```diff
@@ -187,81 +153,6 @@ def dec(f: Any) -> _decorators.PydanticDescriptorProxy[Any]:
     return dec  # type: ignore[return-value]
 
 
-@overload
-def field_validator(
-    __field: str,
-    *fields: str,
-    mode: Literal['before', 'after', 'plain'] = ...,
-    check_fields: bool | None = ...,
-) -> Callable[[_V2BeforeAfterOrPlainValidatorType], _V2BeforeAfterOrPlainValidatorType]:
-    ...
-
-
-@overload
-def field_validator(
-    __field: str,
-    *fields: str,
-    mode: Literal['wrap'],
-    check_fields: bool | None = ...,
-) -> Callable[[_V2WrapValidatorType], _V2WrapValidatorType]:
-    ...
-
-
-FieldValidatorModes: TypeAlias = Literal['before', 'after', 'wrap', 'plain']
-
-
-def field_validator(
-    __field: str,
-    *fields: str,
-    mode: FieldValidatorModes = 'after',
-    check_fields: bool | None = None,
-) -> Callable[[Any], Any]:
-    """
-    Decorate methods on the class indicating that they should be used to validate fields.
-
-    Args:
-        __field (str): The first field the field_validator should be called on; this is separate
-            from `fields` to ensure an error is raised if you don't pass at least one.
-        *fields (tuple): Additional field(s) the field_validator should be called on.
-        mode (FieldValidatorModes): Specifies whether to validate the fields before or after validation.
-             Defaults to 'after'.
-        check_fields (bool | None): If set to True, checks that the fields actually exist on the model.
-            Defaults to None.
-
-    Returns:
-        Callable: A decorator that can be used to decorate a function to be used as a field_validator.
-    """
-    if isinstance(__field, FunctionType):
-        raise PydanticUserError(
-            'field_validators should be used with fields and keyword arguments, not bare. '
-            "E.g. usage should be `@validator('<field_name>', ...)`",
-            code='validator-no-fields',
-        )
-    fields = __field, *fields
-    if not all(isinstance(field, str) for field in fields):
-        raise PydanticUserError(
-            'field_validator fields should be passed as separate string args. '
-            "E.g. usage should be `@validator('<field_name_1>', '<field_name_2>', ...)`",
-            code='validator-invalid-fields',
-        )
-
-    def dec(
-        f: Callable[..., Any] | staticmethod[Any, Any] | classmethod[Any, Any, Any]
-    ) -> _decorators.PydanticDescriptorProxy[Any]:
-        if _decorators.is_instance_method_from_sig(f):
-            raise PydanticUserError(
-                '`@field_validator` cannot be applied to instance methods', code='validator-instance-method'
-            )
-
-        # auto apply the @classmethod decorator
-        f = _decorators.ensure_classmethod_based_on_signature(f)
-
-        dec_info = _decorators.FieldValidatorDecoratorInfo(fields=fields, mode=mode, check_fields=check_fields)
-        return _decorators.PydanticDescriptorProxy(f, dec_info)
-
-    return dec
-
-
 @overload
```

Comment:
`root_validator` should surely move into `validators.py`?

More generally can we remove this module, or move it to `_internal`.

Reply:
I didn't move it because:
1. Backwards compatibility with V1.
2. We want to deprecate `@root_validator` anyway

Reply:
`root_validator` and `validator` where in `class_validators.py` in V1, the logic for supporting imports from the old path is already supported.

I think we move them.